lib/__tests__/findExports-test.js

```diff
@@ -459,3 +439,38 @@ it('does not fail for inner exports that are not objects', () => {
     hasDefault: true,
   });
 });
+
+describe('recursive exports', () => {
+  beforeEach(() => {
+    fs.__setFile(
+      '/path/to/foo.js',
+      'const Result = { bar: 123 }; module.exports = Result;',
+      { isDirectory: () => false },
+    );
+    requireRelative.resolve.mockImplementation(() => '/path/to/foo.js');
+  });
+
+  it('follows exported requires', () => {
+    expect(
+      findExports("module.exports = require('./foo');", '/path/to/file.js'),
+    ).toEqual({
+      named: ['bar'],
+      hasDefault: true,
+    });
+  });
+
+  it('picks the first require if inside a ternary', () => {
+    expect(
+      findExports(
+      `
+        module.exports = process.env.NODE_ENV === 'test' ?
+          require('./foo') : require('./ignored');
```

Comment:
Yeah, it will just stop trying to find exports at that time. I'd assume this is unlikely though, so we can probably roll with this until we have a counterexample.